server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -1314,6 +1314,27 @@ void setDefaultMaxMergesAtOnce(int newDefaultMaxMergesAtOnce) {
         }
     }
 
+    /**
+     * Update the cached defaults for {@code MergeSchedulerConfig.MAX_THREAD_COUNT_SETTING} and
+     * {@code MergeSchedulerConfig.MAX_MERGE_COUNT_SETTING}
+     */
+    void setDefaultMaxThreadAndMergeCount(int maxThreadCount, int maxMergeCount) {
+        // Upon updates to the cluster default settings, we always update the cached default values in
+        // the MergeSchedulerConfig, but we only update the actual values when an index level setting is not present
+        boolean updateActuals = MergeSchedulerConfig.MAX_THREAD_COUNT_SETTING.exists(getSettings()) == false
```

Comment:
This means that if either of max merge count or max thread count index settings are present, then we don't use cluster settings?